lib/dal/src/workspace_snapshot/graph/v4/component.rs

```diff
@@ -168,6 +170,64 @@ impl WorkspaceSnapshotGraphV4 {
             _ => Ok(petgraph::visit::Control::Continue),
         }
     }
+    fn has_socket_connections_dfs_event(
+        event: DfsEvent<NodeIndex>,
+        has_connections: &mut bool,
+        component_id: ComponentId,
+        graph: &Self,
+    ) -> WorkspaceSnapshotGraphResult<Control<()>> {
+        match event {
```

Comment:
This whole thing will be *much much* faster if you just look at every node directly in a single `for node_idx in graph.nodes()`. Again, not sure it matters unless perf is bad, but we should keep an eye out.